### ISSUE TITLE
openvino load tensorflow model.pd error

### ISSUE THREAD

**Issue by linjiaqin** [intel-analytics/analytics-zoo#1703]
2019-10-16 01:11:10 INFO  InferenceSupportive$:45 - optimize tf image classification model execution time elapsed [10 s, 336 ms].
source_0	metal101.jpg	1571159470471	
2019-10-16 01:11:10 INFO  OpenVinoInferenceSupportive$:241 - tf image classification model optimized, log: 
stderr: + [ 8 -ne 8 ]
+ SAVED_MODEL_DIR=/home/linjiaqin/PycharmProjects/tianchitrash/package/savemodel
+ INPUT_SHAPE=[1,224,224,3]
+ REVERSE_INPUT_CHANNELS=1
+ MEAN_VALUES=[123.68,116.78,103.94]
+ SCALE=1.0
+ INPUT=input
+ OPENVINO_IR_OUTPUT_DIR=/tmp/ZooVino4151219403393556622
+ MO_TF_FILE=/tmp/ZooVino6491586636955537423/model-optimizer/mo_tf.py
+ PARMS=--saved_model_dir /home/linjiaqin/PycharmProjects/tianchitrash/package/savemodel --input_shape [1,224,224,3] --input input
+ [ 1 -ne 0 ]
+ echo reverse_input_channels true
+ PARMS=--saved_model_dir /home/linjiaqin/PycharmProjects/tianchitrash/package/savemodel --input_shape [1,224,224,3] --input input --reverse_input_channels
+ echo reverse_input_channels false
+ [ [123.68,116.78,103.94] != [] ]
+ PARMS=--saved_model_dir /home/linjiaqin/PycharmProjects/tianchitrash/package/savemodel --input_shape [1,224,224,3] --input input --reverse_input_channels --mean_value [123.68,116.78,103.94] --scale 1.0
+ PARMS=--saved_model_dir /home/linjiaqin/PycharmProjects/tianchitrash/package/savemodel --input_shape [1,224,224,3] --input input --reverse_input_channels --mean_value [123.68,116.78,103.94] --scale 1.0 --output_dir /tmp/ZooVino4151219403393556622
+ echo /tmp/ZooVino6491586636955537423/model-optimizer/mo_tf.py
+ echo --saved_model_dir /home/linjiaqin/PycharmProjects/tianchitrash/package/savemodel --input_shape [1,224,224,3] --input input --reverse_input_channels --mean_value [123.68,116.78,103.94] --scale 1.0 --output_dir /tmp/ZooVino4151219403393556622
+ python3 /tmp/ZooVino6491586636955537423/model-optimizer/mo_tf.py --saved_model_dir /home/linjiaqin/PycharmProjects/tianchitrash/package/savemodel --input_shape [1,224,224,3] --input input --reverse_input_channels --mean_value [123.68,116.78,103.94] --scale 1.0 --output_dir /tmp/ZooVino4151219403393556622
terminate called after throwing an instance of 'std::bad_alloc'
  what():  std::bad_alloc
Aborted (core dumped)
 
stdout: reverse_input_channels true
reverse_input_channels false
/tmp/ZooVino6491586636955537423/model-optimizer/mo_tf.py
--saved_model_dir /home/linjiaqin/PycharmProjects/tianchitrash/package/savemodel --input_shape [1,224,224,3] --input input --reverse_input_channels --mean_value [123.68,116.78,103.94] --scale 1.0 --output_dir /tmp/ZooVino4151219403393556622
 
exitCode: 134

**this is the input of the model.pd**
MetaGraphDef with tag-set: 'serve' contains the following SignatureDefs:

signature_def['serving_default']:
  The given SavedModel SignatureDef contains the following input(s):
    inputs['input'] tensor_info:
        dtype: DT_FLOAT
        shape: (-1, 224, 224, 3)
        name: conv2d_1_input:0
  The given SavedModel SignatureDef contains the following output(s):
    outputs['output'] tensor_info:
        dtype: DT_FLOAT
        shape: (-1, 6)
        name: activation_4/Softmax:0
  Method name is: tensorflow/serving/predict

**Comment by qiyuangong** [comment]
Hi @linjiaqin 
According to your error message, it seems that your computer cannot allocate more memory for OpenVINO model optimization (when OpenVINO uses TensorFlow to read your model). This error is actually reported by TensorFlow, i.e., https://github.com/tensorflow/tensorflow/issues/9487.

Pls add more memory if you are using virtual machine, or close other applications if you are running on laptop. :)

**Comment by linjiaqin** [comment]
@qiyuangong 
Thank you! But when I solve the above problem. Another error occurred. 

`[ ERROR ]  No node with name input. 
 For more information please refer to Model Optimizer FAQ (<INSTALL_DIR>/deployment_tools/documentation/docs/MO_FAQ.html), question #51. 
 
stdout: reverse_input_channels true
reverse_input_channels false
/tmp/ZooVino2737862234374267838/model-optimizer/mo_tf.py
--saved_model_dir /home/linjiaqin/PycharmProjects/tianchitrash/package/savemodel --input_shape [1,224,224,3] --input input --reverse_input_channels --mean_value [123.68,116.78,103.94] --scale 1.0 --output_dir /tmp/ZooVino7307856848498870606
Model Optimizer arguments:`


```
linjiaqin@linjiaqin-computer:~$ saved_model_cli show --dir /home/linjiaqin/PycharmProjects/tianchitrash/package/savemodel --all

MetaGraphDef with tag-set: 'serve' contains the following SignatureDefs:

signature_def['serving_default']:
  The given SavedModel SignatureDef contains the following input(s):
    inputs['input'] tensor_info:
        dtype: DT_FLOAT
        shape: (-1, 224, 224, 3)
        name: conv2d_1_input:0
  The given SavedModel SignatureDef contains the following output(s):
    outputs['output'] tensor_info:
        dtype: DT_FLOAT
        shape: (-1, 6)
        name: activation_4/Softmax:0
  Method name is: tensorflow/serving/predict
```

**Comment by qiyuangong** [comment]
@linjiaqin 

According to your saved_model_cli, your input node is `conv2d_1_input:0` rather than `input`. So, OpenVINO complain that it cannot find input node.

You can also print node name with code below:
```
import tensorflow as tf
#saved_dir = "./working/saved-model"
saved_dir = "/root/glorysdj/export"
tags = ['serve']
graph = tf.Graph()
sess = tf.Session(graph=graph)
_ = tf.saved_model.loader.load(sess, tags, saved_dir)

for node in graph.as_graph_def().node:
    print(node.name)
```

**Comment by qiyuangong** [comment]
Issue closed. :)
